util: Fix edge case when parsing paths

henne90gen opened the pull request:
Searching for files broke a couple releases ago. It used to be possible to start typing part of a file name, then select a file (not confirm it yet) and then type in `:` and a line number to navigate directly to that line.
The current behavior can be seen in the following screenshots. When the `:` is typed, the selection is lost, since no files match any more.
<img width="552" height="370" alt="Screenshot From 2025-08-12 10-36-08" src="https://github.com/user-attachments/assets/e4b4b613-7f0c-40d7-94c9-04d8ab541656" />
<img width="553" height="124" alt="Screenshot From 2025-08-12 10-36-25" src="https://github.com/user-attachments/assets/843e9ecf-9e08-4fa6-9340-0388a957cbb2" />
<img width="549" height="370" alt="Screenshot From 2025-08-12 10-36-47" src="https://github.com/user-attachments/assets/4a1bbbd8-268a-4ea8-999f-6cef1eb34a45" />

---

With this PR, the previous behavior is restored and can be seen in these screenshots:

<img width="552" height="370" alt="Screenshot From 2025-08-12 10-36-08" src="https://github.com/user-attachments/assets/466e1906-4735-47ae-a699-117bdd6490ca" />
<img width="549" height="370" alt="Screenshot From 2025-08-12 10-47-07" src="https://github.com/user-attachments/assets/17f3acda-662d-4962-9eb8-4b494f211d26" />
<img width="549" height="370" alt="Screenshot From 2025-08-12 10-47-21" src="https://github.com/user-attachments/assets/d98447fe-7377-4f4f-b3da-f690cd44c141" />

---

Release Notes:

- Adjusted the file finder to show matching file paths when adding the `:row:column` to the query

henne90gen committed:
util: Fix edge case when parsing paths

henne90gen commented:
@cla-bot check

SomeoneToIgnore reviewed (dismissed):
This does not look like a good place to deal with such things: `:` is a valid char in Unix paths, and `PathWithPosition::parse_str` can be used way past path canonicalization, where we could have established that `/foo/bar:aa` is a fully valid path.

Seems that the fix would better belong to `file_finder` and `file_finder_tests` modules instead: there, we could parse the query and, if the query had got some lines/columns, keep the selection.

henne90gen commented:
I've moved the fix into `file_finder`. I'll update the tests as soon as possible.

henne90gen commented:
@SomeoneToIgnore could you take another look at this PR please? I've updated the code and also updated one of the test cases to ensure this does not break again.

SomeoneToIgnore commented on crates/file_finder/src/file_finder.rs:
```diff
@@ -1403,11 +1403,14 @@ impl PickerDelegate for FileFinderDelegate {
             let raw_query = raw_query.trim().to_owned();
 
-            let file_query_end = if path_position.path.to_str().unwrap_or(&raw_query) == raw_query {
+            let raw_query = raw_query.trim_end_matches(':').to_owned();
```
This looks odd a bit, as technically does not cover edge-cases properly (however niche and edgy they are):

1. 
```
src/
      foo:important.rs
      foo.rs
```

for the `foo:` query should return a single file.

The new code seem to be wrongfully returning 2 now?
That better have a test.


2. Besides, it's somewhat odd to allocate twice? 
It seems that could call `to_owned` once on a resulting query, not on each transformation.

3.

`Some(path.unwrap().trim_end_matches(':').len())` is quite confusing to me — we take a __real__ path that had been returned and spoil it? 
What for?

henne90gen replied:
I've taken care of point 2. and 3. (you were right, `path` did not need to be trimmed).
Your first point is a bit more tricky, since we cannot differentiate between the colon being part of the file name and the colon being part of the row/column declaration. When typing `foo:` I might want to jump to `foo:important.rs`, or I might want to jump to a specific row inside `foo.rs`. In my opinion it would be better to show more results and thus fix the bug I've mentioned in the description of this PR.

SomeoneToIgnore commented on crates/util/src/paths.rs:
```diff
@@ -449,6 +449,7 @@ impl PathWithPosition {
                         path_string.push_str(maybe_row);
                     }
                 }
+
```
Not needed.

SomeoneToIgnore reviewed (approved):
Thank you, with the test it is much more clear what happens and that looks reasonable.

CnsMaple commented:
@henne90gen Sorry, I want to know how to select files by number now.I am using Windows 10.Zed Dev 0.202.0 
f8667a837949597200e2ae8e490d947c6cda75aa
<img width="1330" height="527" alt="0b5va0sf otx" src="https://github.com/user-attachments/assets/77e52920-0100-4512-8793-b0b56177234e" />
